src/runtime_src/core/common/device.h

```diff
@@ -109,6 +109,7 @@ class device : public ishim
   using slot_id = xclbin_map::slot_id;
   using handle_type = xclDeviceHandle;
   using memory_type = xrt::xclbin::mem::memory_type;
+  std::mutex device_lock;
```

Comment:
Move to private data members, this mutex should definitely not be public.

Reply:
Done